### ISSUE TITLE
IDEATION: UX insights for reply threads

### ISSUE THREAD

**Issue by BeccaMelhuish** [citizenos/citizenos-fe#1249]
@kevincrepin @ilmartyrk 

**Round 1 ideation admin UX test insights:** 

A few insights from the admin tests - some stronger than others:  

- It seemed fairly solid that the users didn't need/want to have to put a heading on their replies (like we don't in discussions). One user in particular got really confused (took a lot of time to understand the error messages) and a bit frustrated by having to fill in both, actually it was her main 'point for improvement' in her summary at the end as it had bugged her that much :D 

![image](https://github.com/citizenos/citizenos-fe/assets/61320526/813480ca-e0f8-47d5-82bd-65ea7aa1ac49)

- One user suggested the replies should work more like they do in Facebook (and so more like they do in our discussion replies, though she didn't know that) - so that it's clearer which reply you are replying to, or to the idea itself. 
- Another user said similar, that it should be possible to reply directly to a reply - to him each reply "looks like a new thread". 

- One user didn't see my reply to her reply, and said "Let me re-load as I don't see it at the moment', when I could see it clearly on her screen (screenshot below). I'm not sure, it may have been because my name was cropped off in her view (bug issue created [here](url)), but I believe this may have been because she was expecting it to show up underneath her reply, rather than above it. 

![image](https://github.com/citizenos/citizenos-fe/assets/61320526/e8b3b2e0-3938-4255-9dcf-4e81fb625eaf)

- In the discussion reply threads, one user showed surprise and confusion at my name appearing in front of his reply to me, once it was posted. He said that he read it as if it was saying I had written the reply, rather than him. (Though this may be because his profile icon was white, so more easy to miss his own name above it). 

![image](https://github.com/citizenos/citizenos-fe/assets/61320526/19c123ba-e239-4005-9e0b-0385b873077d)

- Re the above point, several of the users (admin and participant) didn't get to see how their reply posted (in either ideation or discussion), so I couldn't really get insight on this, as the reply thread closes once it has been posted (so they didn't see the name appearing at the front of their reply. 

**Round 1 ideation participant UX test insights:** 

- The issue of reply thread structure was raised as a problem in the participant tests too - that they want conversation to happen "but I'm not able to freely respond to this individually" meaning reply directly to a reply. They see it as being posted separately. 

- One participant praised that the discussion threads work very nicely, with the reply structure of who has replied to who. 

- One participant explained (when we were at the end of the ideation phase) that her expectation of what might happen in the discussion phases is that the pros and cons of the ideas get discussed. Or of certain ideas that need more discussion. 

**Suggestions:** 

- Make the reply threads work the same way as they do on discussion threads, so that: 

(1) No heading needed for a reply, you just write a reply like in discussions

(2) You can clearly reply to different replies, and see which reply connects to which reply, same as we have in discussions. 

(3) Have the replies appear open not collapsed, so the users can see them, after they have written them - separate issue [here](https://github.com/citizenos/citizenos-fe/issues/1263). (Not solid insights for this, but I think is more standard UX and makes sense they'd want to see it, do you think @kevincrepin?) 

(4) The reply thread loads downwards, first reply at the top then later replies underneath. (No solid insight for this, just a hunch as above, plus to me it seems more logical and consistent with the discussion threads). 

(5) Work on the way the name appears in front of the user's reply, so that it (a) appears as they are writing the reply (so re this [issue here] (https://github.com/citizenos/citizenos-fe/issues/1005)) not only once posted, and also (b) so that it's a bit clearer who is writing the message? This is how facebook does it, which is close to ours, but maybe there is a subtle change we can make to make it clearer? This latter (point b) issue was only raised one user though, so we can also just leave it and keep an eye in tests once replies land "open" not collapsed (point 3 above). 
![image](https://github.com/citizenos/citizenos-fe/assets/61320526/d5225f86-ef01-4fcd-b165-f8e7a937def9)

(6) Something we might want to consider is whether we want to have pro and con replies underneath ideas. Some reasons/thoughts on this:

*As above, the final participant mentioned an expectation that this might happen in the discussion phase (which shows she felt there was a need for it).

*From watching the other tests, it had already seemed the idea replies often fit into this pro/con structure (e.g. more or less "Good idea!" + reason/additional point / "I disagree" + reason/additional point). The pro and con set up actually seems to fit better here than in most discussions I've seem in the tests so far. 

*Another (admin) user was talking about using the discussion section to discuss whether to go ahead with ideas that didn't get consensus, when there is already a discussion section under the ideas themselves. So maybe (though he didn't say so) the pro-con discussion format in the discussion phases seems more suitable for this than the replies under the ideas. 

*Maybe something to just keep an eye on though, and see with more testing if it's wanted/needed / see what happens with the discussion phase prompt idea. Feels like it might over complicate the ideation part a bit..

**Comment by BeccaMelhuish** [comment]
I'd say this is mostly an **Ideation launch fixes** one if possible? For whichever of the suggestions are taken forward, but I'd say definitely making it work as our discussion threads already do?

**Comment by kevincrepin** [comment]
**EDIT: updated screenshots according to fixes due to later issues:**

If we can take over the threading from the discussions, then I would say we also need to move the replies from the side to under the idea. This also immediately sets it up better for later, because both contents are less confined to their widths.

Otherwise I would say let's take over everything we can from the discussion:

1. Hide replies by default
![Screenshot 2024-06-07 at 15 10 10](https://github.com/citizenos/citizenos-fe/assets/26792281/bb54cc42-74bb-4921-b5f7-0a433b5291de)


2. View threaded replies
![Screenshot 2024-06-07 at 15 10 20](https://github.com/citizenos/citizenos-fe/assets/26792281/7b0c01ab-602b-4f38-8e9c-7b9ebffe9379)


@ilmartyrk @BeccaMelhuish there are two ways we can do the scrolling as it can expand quite a lot:
v1: https://www.figma.com/proto/gOJ4i00538e0qaP6IKIPVE/Platform-2.0---Design?page-id=3094%3A111670&node-id=13338-116996&viewport=-17953%2C-14491%2C0.5&t=Tl5iu6mC1F17FuzI-1&scaling=scale-down&starting-point-node-id=13338%3A116996&show-proto-sidebar=1
v2: https://www.figma.com/proto/gOJ4i00538e0qaP6IKIPVE/Platform-2.0-Design?page-id=3094%3A111670&node-id=13338-117174&viewport=-17953%2C-14491%2C0.5&t=Tl5iu6mC1F17FuzI-1&scaling=scale-down&starting-point-node-id=13338%3A117174&show-proto-sidebar=1

My preference is v1, but it's quite subjective.

**Comment by BeccaMelhuish** [comment]
Wonderful yes this looks great, and better for when we have bigger space for replies like you say.

Preference for v1 too! More breathing space to view the content :) 

Re replies collapsed by default, what are your thoughts on [this though](https://github.com/citizenos/citizenos-fe/issues/1263) - they should be stay open after someone has posted their reply right? (Watching the users in tests, it definitely feels like it should be).

**Comment by kevincrepin** [comment]
@ilmartyrk I added now also mobile views: https://www.figma.com/design/gOJ4i00538e0qaP6IKIPVE/Platform-2.0---Design?node-id=13338-115399&t=5P8GCGSpqoB12ZBy-1

One addiitonal change I would like to request if ok by you and @BeccaMelhuish. I repositioned the posting time to under the user's name:
![Screenshot 2024-06-18 at 11 20 01](https://github.com/citizenos/citizenos-fe/assets/26792281/c810ef3d-0997-4331-aad3-18b019678d9b)

Previously it was aligned to the right next to the actions button. However in mobile the left and right labels would start to merge into each other. This way it would be easier to get the same layout in mobile as we have in desktop and stay recognisable.

I'm thinking also the same could be done for the discussions? Check below how it is currently:
![Screenshot 2024-06-18 at 11 21 51](https://github.com/citizenos/citizenos-fe/assets/26792281/2625146f-6ec1-4a33-9146-8cb467aca15b)

And updated according to the same logic as in Ideation:
![Screenshot 2024-06-18 at 11 22 55](https://github.com/citizenos/citizenos-fe/assets/26792281/3efc8b94-4fc4-448b-b860-b2a6244f7c6f)

**Comment by BeccaMelhuish** [comment]
@ilmartyrk not a priority to fix before testing, but some issues with the number of replies not showing properly: 

- There are 5 but it says 6. I'm not sure which way around the problem is (that I posted 6 and one isn't showing, or that I only posted 5 but it says 6). 

![image](https://github.com/citizenos/citizenos-fe/assets/61320526/16c1b0c5-d0f1-400b-86b3-631eab0c57c6)

It says 2 here: 

![image](https://github.com/citizenos/citizenos-fe/assets/61320526/e7f0db19-86e2-49eb-929f-d8c9102ef9a4)

But there are 5 in total. Though this I am unsure about. To me it seems that it should say 5, but maybe @kevincrepin has a view on this - should all the replies including replies to replies be included in the top level? 

![image](https://github.com/citizenos/citizenos-fe/assets/61320526/f949feba-02c1-4ca0-856d-03ddb8e3c875)

**Comment by ilmartyrk** [comment]
@BeccaMelhuish fixed the numbers

**Comment by BeccaMelhuish** [comment]
Grand, seems good to me but will leave open in case @kevincrepin thinks it should be otherwise or @ssin1901 needs to test any more :)

**Comment by ilmartyrk** [comment]
@BeccaMelhuish @ssin1901 added the mobile view
